doc/users/interactive_guide.rst

```diff
@@ -0,0 +1,454 @@
+
+For example, if you want to poll for data you could use something like ::
+
+  fig, ax = plt.subplots()
+  ln, = ax.plot([], [])
+
+  while True:
+      x, y = get_new_data()
+      ln.set_data(x, y)
+      fig.canvas.draw_idle()
+      plt.pause(1)
+
+which would poll for new data and update the figure at 1Hz.
+
+.. _spin_event_loop:
+
+Explicitly spinning the Event Loop
+----------------------------------
+
+.. autosummary::
+   :template: autosummary.rst
+   :nosignatures:
+
+   backend_bases.FigureCanvasBase.flush_events
+   backend_bases.FigureCanvasBase.draw_idle
+
+
+
+If you have open windows that have pending UI
+events (mouse clicks, button presses, or draws) you can explicitly
+process those events by calling `.FigureCanvasBase.flush_events`.
+This will run the GUI event loop until all UI events currently waiting
+have been processed.  The exact behavior is backend-dependent but
+typically events on all figure are processed and only events waiting
+to be processed (not those added during processing) will be handled.
+
+For example ::
+
+   import time
+   import matplotlib.pyplot as plt
+   import numpy as np
+   plt.ion()
+
+   fig, ax = plt.subplots()
+   fig.canvas.show()
+   th = np.linspace(0, 2*np.pi, 512)
+   ax.set_ylim(-1.5, 1.5)
+
+   ln, = ax.plot(th, np.sin(th))
+
+   def slow_loop(N, ln):
+       for j in range(N):
+           time.sleep(.1)  # to simulate some work
+           ln.figure.canvas.flush_events()
+
+   slow_loop(100, ln)
+
+While this will feel a bit laggy (as we are only processing user input
+every 100ms where as 20-30ms is what feels "responsive") it will
+respond.
+
+If you make changes to the plot and want it re-rendered you will need
+to call `~.FigureCanvasBase.draw_idle` to request that the canvas be
+re-drawn.  This method can be thought of *draw_soon* in analogy to
+`asyncio.loop.call_soon`.
+
+We can add this our example above as ::
+
+   def slow_loop(N, ln):
+       for j in range(N):
+           time.sleep(.1)  # to simulate some work
+           if j % 10:
+               ln.set_ydata(np.sin(((j // 10) % 5 * th)))
+               ln.figure.canvas.draw_idle()
+
+           ln.figure.canvas.flush_events()
+
+   slow_loop(100, ln)
+
+
+The more frequently you call `.FigureCanvasBase.flush_events` the more
+responsive your figure will feel but at the cost of spending more
+resource on the visualization and less on your computation.
+
+
+.. _stale_artists:
+
+Stale Artists
+=============
+
+Artists (as of Matplotlib 1.5) have a **stale** attribute which is
+`True` if the internal state of the artist has changed since the last
+time it was rendered. By default the stale state is propagated up to
+the Artists parents in the draw tree, thus if the color of a `.Line2D`
+instance is changed, the `.axes.Axes` and `.figure.Figure` it is
+contained in will also be marked as "stale".  Thus, ``fig.stale`` will
+report of any artist in the figure has been modified and out of sync
```

Comment:
```suggestion
time it was rendered. By default the stale state is propagated up to
the Artists parents in the draw tree, e.g., if the color of a `.Line2D`
instance is changed, the `.axes.Axes` and `.figure.Figure` that
contain it will also be marked as "stale".  Thus, ``fig.stale`` will
report if any artist in the figure has been modified and is out of sync
```